Formula/vala-language-server.rb

```diff
@@ -0,0 +1,31 @@
+class ValaLanguageServer < Formula
+  desc "Code Intelligence for Vala & Genie"
+  homepage "https://github.com/Prince781/vala-language-server"
+  url "https://github.com/Prince781/vala-language-server/archive/3e01b8383b3db3c39af276528663d6084c671455.tar.gz"
+  version "0.48.3"
+  sha256 "6f3b34bcb4e049c299ae3d5433153e4b685b0bace0ea8d761ffea266714ce841"
+  license "LGPL-2.1-only"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "vala" => :build
+  depends_on "glib"
+  depends_on "json-glib"
+  depends_on "jsonrpc-glib"
+  depends_on "libgee"
+
+  def install
+    system "meson", "-Dplugins=false", "-Dprefix=#{prefix}", "build"
+    system "ninja", "-C", "build"
+    system "ninja", "-C", "build", "install"
+  end
+
+  test do
+    # There are currently no tests for VLS
```

Comment:
This shouldn't matter, as we don't want upstream's tests to be used in the `test` block. Instead, you should try to simulate a user interacting with `vala-language-server` and check that it behaves correctly.

You might want to look at the other language servers in Homebrew/core for examples.

Reply:
I tried to write something similar, but, if I'm not mistaken, you can only communicate with VLS via the JSON-RPC protocol. VLS only takes `--help` or `--version` as an argument. The upcoming official tests just use the JSON-RPC protocol. When they are ready, it will be easy to write the test. Now, with my knowledge, there is no other way to do it than using just these two arguments.

Reply:
So users can't run anything other than `--help` or `--test`? Seems like a pretty unhelpful tool then.

Reply:
VLS is built to communicate with IDEs and editors, not the user. There it automatically completes and checks the Vala syntax. The user uses it through the IDE, not the console, and the IDE communicates with it just via the JSON-RPC protocol mentioned above.

Before I started creating this formula, VLS didn't even have `--help` and `--verison` because the user didn't need it. These arguments were added just for the sake of the test in this formula.

Reply:
How does VLS communicate with the IDE? Can't we mock an IDE for VLS to interact with?

Reply:
That is what they are trying to do in the official test: https://github.com/Prince781/vala-language-server/blob/master/test/testclient.vala

Reply:
The testclient merely checks the server can respond to the initialize request. I'm working on a framework for writing scripts that simulate an editor interacting with a language server [here](https://GitHub.com/Prince781/lstf), but it's not ready yet.

Reply:
So there aren't any tests for now, but if it's absolutely necessary I could write some simple ones.

Reply:
If it's json-rpc, can't we just:

1. fire up the server
1. Send it a request with cURL 
1. See that it fails as expected

Reply:
You could pipe some JSON to it and check for the right output, yeah.

Reply:
> You could pipe some JSON to it and check for the right output, yeah. Socket connections aren't supported.

That's pretty much what all the other language server formulae do. @pervoj suggested this was not possible for some reason after I suggested mimicking the tests of these formulae.

Reply:
> That's pretty much what all the other language server formulae do. @pervoj suggested this was not possible for some reason after I suggested mimicking the tests of these formulae.

@carlocab Yes, I didn't understand, how does the tests in other LS formulae work. Now I'm trying to write some JSON, which will work. So sorry for that.

Reply:
Done, now it should work.

Reply:
Should I edit it again to pass the bot test?